File: demo-web-scala/src/main/scala/com/example/demo/security/WebSecurityConfig.scala
package com.example.demo.security

import org.springframework.context.annotation.{Bean, Configuration}
import org.springframework.security.config.annotation.web.builders.HttpSecurity
import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter
import org.springframework.security.crypto.password.{NoOpPasswordEncoder, PasswordEncoder}

@Configuration
class WebSecurityConfig extends WebSecurityConfigurerAdapter {

  @Bean
  override def userDetailsService = new CustomUserDetailsService()

  @throws[Exception]
  override protected def configure(http: HttpSecurity): Unit = {
    http.csrf().disable()
      .formLogin().loginPage("/login").permitAll().and()
      .authorizeRequests.antMatchers("/resources/**", "/storage/**", "/login").permitAll
      .antMatchers("/api/**").hasAuthority("USER")
      .antMatchers("/admin/**").hasAuthority("ADMIN")
      .anyRequest.authenticated()
  }

  @Bean
  def passwordEncoder: PasswordEncoder = NoOpPasswordEncoder.getInstance
}

File: demo-web-java/src/main/java/com/example/demo/security/WebSecurityConfig.java
package com.example.demo.security;

import com.example.demo.model.Authority;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.security.config.annotation.web.builders.HttpSecurity;
import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
import org.springframework.security.crypto.password.NoOpPasswordEncoder;
import org.springframework.security.crypto.password.PasswordEncoder;

@Configuration
public class WebSecurityConfig extends WebSecurityConfigurerAdapter {
  @Bean
  public CustomUserDetailsService userDetailsService() {
    return new CustomUserDetailsService();
  }
  protected void configure(HttpSecurity http) throws Exception {
    http
        .authorizeRequests()
          .antMatchers("/resources/**", "/storage/**").permitAll()
          .antMatchers("/").hasAnyAuthority(Authority.ADMIN.name(), Authority.USER.name())
        .anyRequest().authenticated()
        .and()
        .formLogin().loginPage("/login").permitAll().and()
        .csrf().disable();
  }
  @Bean
  public PasswordEncoder passwordEncoder() {
    return NoOpPasswordEncoder.getInstance();
  }
}
